rskj-core/src/test/java/co/rsk/mine/MinerServerTest.java

```diff
@@ -651,6 +688,96 @@ public void extraDataWithClientDataMoreThan32Bytes() {
         assertEquals("tincho is the king of mining".substring(0, clientExtraDataSize), new String(thirdItem));
     }
 
+    @Test
+    public void onBestBlockBuildBlockToMine() {
+
+        // prepare for miner server
+        Ethereum ethereum = mock(EthereumImpl.class);
+        when(ethereum.addNewMinedBlock(any())).thenReturn(ImportResult.IMPORTED_BEST);
+
+        BlockProcessor blockProcessor = mock(NodeBlockProcessor.class);
+        when(blockProcessor.hasBetterBlockToSync()).thenReturn(false);
+        
+        BlockUnclesValidationRule unclesValidationRule = mock(BlockUnclesValidationRule.class);
+        when(unclesValidationRule.isValid(any())).thenReturn(true);
+        
+        MinerClock clock = new MinerClock(true, Clock.systemUTC());
+        
+        // create miner server
+        MinerServer minerServer = spy(makeMinerServer(ethereum, blockProcessor, unclesValidationRule, clock, transactionPool));
+        minerServer.start();
+   
+        // create listener
+        MinerServerImpl.NewBlockTxListener listener = new MinerServerImpl.NewBlockTxListener(this.miningMainchainView, minerServer, blockProcessor, true);
```

Comment:
Now that I see it, we already have one, you had to add `times(2)` in line 721 due to this.

Reply:
As a quick thought, a more or less easy way to avoid this is to inject into `MinerServerImpl` a factory that creates this listener. This way you can mock this factory `create()` call from the tests, make it return the listener you create here, and verify on it.

This way, IMO, tests get easier and we are not coupled to this listener (`NewBlockTxListener` should no longer be inner in this case).

Reply:
1- the two instances thing has no side effects
2- the times(2) its because the buildBlock is called once in the start and the second time on the event
3- this would be too much refactoring for now

Reply:
sure, it was just an idea, and I understand you don't want to do this refactor now 👍